aws/resource_aws_network_acl_association.go

```diff
@@ -0,0 +1,140 @@
+	conn := meta.(*AWSClient).ec2conn
+
+	naclId := d.Get("network_acl_id").(string)
+	subnetId := d.Get("subnet_id").(string)
+
+	log.Printf(
+		"[INFO] Creating network acl association: %s => %s",
+		subnetId,
+		naclId)
+
+	association, err_association := findNetworkAclAssociation(subnetId, conn)
+	if err_association != nil {
+		return fmt.Errorf("Failed to create acl %s with nacl %s: %s", d.Id(), naclId, err_association)
+	}
+
+	associationOpts := ec2.ReplaceNetworkAclAssociationInput{
+		AssociationId: association.NetworkAclAssociationId,
+		NetworkAclId:  aws.String(naclId),
+	}
+
+	var err error
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err = conn.ReplaceNetworkAclAssociation(&associationOpts)
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
+				if awsErr != nil {
+					return resource.RetryableError(awsErr)
+				}
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Set the ID and return
+	d.SetId(naclId)
+	log.Printf("[INFO] Association ID: %s", d.Id())
+
+	return resourceAwsNetworkAclAssociationRead(d, meta)
+}
+
+func resourceAwsNetworkAclAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	// Inspect that the association exists
+	subnetId := d.Get("subnet_id").(string)
+	_, err_association := findNetworkAclAssociation(subnetId, conn)
+	if err_association != nil {
+		log.Printf("[INFO] Association for %s was not found.", subnetId)
+		d.SetId("")
+		return nil
+	}
+
+	return nil
+}
+
+func resourceAwsNetworkAclAssociationUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	naclId := d.Get("network_acl_id").(string)
+	subnetId := d.Get("subnet_id").(string)
+
+	log.Printf(
+		"[INFO] Creating network acl association: %s => %s",
+		subnetId,
+		naclId)
+
+	association, err_association := findNetworkAclAssociation(subnetId, conn)
+	if err_association != nil {
+		return fmt.Errorf("Failed to update acl %s with subnet %s: %s", d.Id(), naclId, err_association)
+	}
+
+	req := &ec2.ReplaceNetworkAclAssociationInput{
+		AssociationId: association.NetworkAclAssociationId,
+		NetworkAclId:  aws.String(naclId),
+	}
+	resp, err := conn.ReplaceNetworkAclAssociation(req)
+
+	if err != nil {
+		ec2err, ok := err.(awserr.Error)
+		if ok && ec2err.Code() == "InvalidAssociationID.NotFound" {
+			// Not found, so just create a new one
+			return resourceAwsNetworkAclAssociationCreate(d, meta)
+		}
+
+		return err
+	}
+
+	// Update the ID
+	d.SetId(*resp.NewAssociationId)
+	log.Printf("[INFO] Association ID: %s", d.Id())
+
+	return resourceAwsNetworkAclAssociationRead(d, meta)
+}
+
+func resourceAwsNetworkAclAssociationDelete(d *schema.ResourceData, meta interface{}) error {
+
+	log.Printf("[INFO] Do nothing on network acl associatioØ destroy phase: %s", d.Id())
```

Comment:
I think this method should really do something here, as the association would still exist in the AWS-land. 
What do you think of doing something in the idea of: replacing the Network ACL Association created with the default Network ACL, as it is done in the [resource_aws_network_acl.go](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_network_acl.go#L488) resource?

Also, when you need to destroy something in the TF-land, set the ID to `""` and it will be remove from the state, like `d.SetId("")`

Reply:
Thank you for comment
I think too